apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloCacheInterceptor.java

```diff
@@ -124,7 +125,9 @@ InterceptorResponse resolveFromCache(InterceptorRequest request) throws ApolloEx
   Set<String> cacheResponse(final InterceptorResponse networkResponse,
       final InterceptorRequest request) {
     if (networkResponse.parsedResponse.isPresent() && networkResponse.parsedResponse.get().hasErrors()) {
-      return Collections.emptySet();
+      if (!request.cacheHeaders.hasHeader(ApolloCacheHeaders.STORE_PARTIAL_RESPONSES)) {
```

Comment:
Nit: maybe merge this condition with the one above? A second `if` isn't really required here.

Reply:
Done. I hope formatting is fine, IntelliJ doesn't seem to format the file for me at all 😕

Reply:
Looks good, thanks!

Reply:
Damn, checkstyle is now complaining with: `'&&' should be on a new line.`. Can you move that `&&`?